src/optimizer/OptimizerUtils.cpp

```diff
@@ -445,6 +445,9 @@ Value OptimizerUtils::normalizeValue(const meta::cpp2::ColumnDef& col, const Val
                 col.get_type().get_type_length() == nullptr) {
                 return Value::kNullBadType;
             }
+            if (!v.isStr()) {
+                return v;
```

Comment:
return Value::kNullBadType?

Reply:
> return Value::kNullBadType?

This function is just to normalize the value in the opt phase.For relational expressions, comparison of different types is also a normal input scenario.